components/gfx/display_list/mod.rs

```diff
@@ -1436,11 +1410,13 @@ impl<'a> Iterator for DisplayItemIterator<'a> {
 
 impl DisplayItem {
     /// Paints this display item into the given painting context.
-    fn draw_into_context(&self, paint_context: &mut PaintContext) {
-        let this_clip = &self.base().clip;
-        match paint_context.transient_clip {
-            Some(ref transient_clip) if transient_clip == this_clip => {}
-            Some(_) | None => paint_context.push_transient_clip((*this_clip).clone()),
+    fn draw_into_context(&self, transform: &Matrix4, paint_context: &mut PaintContext) {
```

Comment:
Sure. ```transform``` is necessary to properly draw StackingContexts, because it is used to calculate a new transform based on the StackingContexts bounds (because display items are positioned relative to stacking contexts). Before StackingContexts were drawn outside of DisplayItem::draw_into_context (in DisplayList::draw_into_context). Now that StackingContexts can be DisplayItems it is moved here and so the transform must be included in the case that this item is a StackingContext item.